### PR TITLE
ascanrulesBeta: update reference from http to https 

### PR DESCRIPTION
**Pull request by PariharAditya**
Signed-off-by: Aditya Parihar <97739299+PariharAditya@users.noreply.github.com>

## Overview
ascanrulesBeta 
https://www.zaproxy.org/docs/alerts/40044/
## Related Issues
zaproxy/zaproxy#8262
## Checklist
- [ ] Update help
- [x] Update changelog
- [ ] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [ ] Squash commits
- [ ] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by thc202** [comment]
You need to rebase your branch.

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Changed
 - Update minimum ZAP version to 2.14.0.
 - Update references for Expression Language Injection and HTTP Parameter Pollution (Issue 8262).
+  - Exponential Entity Expansions
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/CHANGELOG.md:10]: Combine this with the previous one.

File: addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties [hunk 2]
```diff
@@ -55,7 +55,7 @@ ascanbeta.elinjection.soln = Perform data validation best practice against untru
 ascanbeta.entityExpansion.desc = An exponential entity expansion, or "billion laughs" attack is a type of denial-of-service (DoS) attack. It is aimed at parsers of markup languages like XML or YAML that allow macro expansions.
 ascanbeta.entityExpansion.name = Exponential Entity Expansion (Billion Laughs Attack)
 ascanbeta.entityExpansion.other = The attack HTTP request received a response after {0} seconds.
-ascanbeta.entityExpansion.refs = https://en.wikipedia.org/wiki/Billion_laughs_attack\nhttp://projects.webappsec.org/XML-Entity-Expansion\nhttp://cwe.mitre.org/data/definitions/776.html
+ascanbeta.entityExpansion.refs = https://en.wikipedia.org/wiki/Billion_laughs_attack\nhttp://projects.webappsec.org/XML-Entity-Expansion\nhttps://cwe.mitre.org/data/definitions/776.html
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties:58]: You missed one 😉

File: addOns/ascanrulesBeta/CHANGELOG.md (hunk 1 above)
Comment (thc202) [reply addOns/ascanrulesBeta/CHANGELOG.md:10]: There's really no previous one, this was already released.

Comment (kingthorin) [reply addOns/ascanrulesBeta/CHANGELOG.md:10]: I meant the line right above. Or is this missing a rebase?

Comment (thc202) [reply addOns/ascanrulesBeta/CHANGELOG.md:10]: Missing a rebase, this is no longer part of the Unreleased section.

Comment (PariharAditya) [reply addOns/ascanrulesBeta/CHANGELOG.md:10]: so i don't need to make changes here ?

Comment (thc202) [reply addOns/ascanrulesBeta/CHANGELOG.md:10]: You need to rebase first and then move "this" (with changes) to the Unreleased section.

File: addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties [hunk 3]
```diff
@@ -55,7 +55,7 @@ ascanbeta.elinjection.soln = Perform data validation best practice against untru
 ascanbeta.entityExpansion.desc = An exponential entity expansion, or "billion laughs" attack is a type of denial-of-service (DoS) attack. It is aimed at parsers of markup languages like XML or YAML that allow macro expansions.
 ascanbeta.entityExpansion.name = Exponential Entity Expansion (Billion Laughs Attack)
 ascanbeta.entityExpansion.other = The attack HTTP request received a response after {0} seconds.
-ascanbeta.entityExpansion.refs = https://en.wikipedia.org/wiki/Billion_laughs_attack\nhttp://projects.webappsec.org/XML-Entity-Expansion\nhttp://cwe.mitre.org/data/definitions/776.html
+ascanbeta.entityExpansion.refs = https://en.wikipedia.org/wiki/Billion_laughs_attack\nhttps://cse.google.com/cse?cx=partner-pub-9396229490951644%3A4ygj091ckzs&ie=ISO-8859-1&q=XML+Entity+Expansion&sa=Search&siteurl=www.webappsec.org%2F&ref=www.google.com%2F&ss=125j15625j2\nhttps://cse.google.com/cse?cx=partner-pub-9396229490951644%3A4ygj091ckzs&ie=ISO-8859-1&q=XML+Entity+Expansion&sa=Search&siteurl=www.webappsec.org%2F&ref=www.google.com%2F&ss=125j15625j2\nhttps://cwe.mitre.org/data/definitions/776.html
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties:58]: You can just drop the middle link if it isn't available via https. I'm not confident that google cache links are stable.

You could also use: https://owasp.org/www-community/vulnerabilities/XML_External_Entity_(XXE)_Processing

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 4]
```diff
@@ -4,9 +4,8 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
-
-## [49] - 2024-01-16
```
Comment (thc202) [inline addOns/ascanrulesBeta/CHANGELOG.md:9]: This should be kept.

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 5]
```diff
@@ -4,9 +4,8 @@ All notable changes to this add-on will be documented in this file.
...
+- Update references for Expression Language Injection and HTTP Parameter Pollution (Issue 8262).
+  - Exponential Entity Expansion
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/CHANGELOG.md:8]: This still isn't right. The "new" change should be under a `### Changed` heading in the Unreleased section.

```md
### Changed
- Update references for Exponential Entity Expansion (Issue 8262).
```

Line 7 is no longer needed as it was in the changelog and released as part of v49,

**Comment by kingthorin** [comment]
I'm at a loss as to how to explain this any differently or further.

**Comment by PariharAditya** [comment]
> I'm at a loss as to how to explain this any differently or further.

sorry sir

**Comment by kingthorin** [comment]
The changelog should look like this:

```md
# Changelog
All notable changes to this add-on will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

## Unreleased
### Changed
- Update references for the Exponential Entity Expansion (Billion Laughs Attack) scan rule (Issue 8262).

## [49] - 2024-01-16
### Changed
- Update minimum ZAP version to 2.14.0.
- Update references for Expression Language Injection and HTTP Parameter Pollution (Issue 8262).
- The Source Code Disclosure - SVN scan rule includes example alert functionality for documentation generation purposes (Issue 6119).

### Removed
- Help entry for the Spring Actuators scan rule (missed during previous removal/promotion).

## [48] - 2023-09-08
### Added
- The HTTP Only Site scan rule now includes example alert functionality for documentation generation purposes (Issue 6119).
```

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 6]
```diff
@@ -4,10 +4,10 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
+### Changed
+- Update references for the Exponential Entity Expansion (Billion Laughs Attack) scan rule (Issue 8262).
 
 ## [49] - 2024-01-16
-### Changed
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/CHANGELOG.md:10]: Restore this one and it should be on the mark 👍

Comment (PariharAditya) [reply addOns/ascanrulesBeta/CHANGELOG.md:10]: done

File: addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties (hunk 3 above)
Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties]: Okay, now to the actual work at hand 🙂

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by PariharAditya** [comment]
> Thank you!

🙇‍♂️

**Pull request merged**